delta/sourcing/src/main/scala/ch/epfl/bluebrain/nexus/sourcing/processor/StopStrategyConfig.scala

```diff
@@ -0,0 +1,37 @@
+package ch.epfl.bluebrain.nexus.sourcing.processor
+
+import ch.epfl.bluebrain.nexus.sourcing.processor.StopStrategy.{PersistentStopStrategy, TransientStopStrategy}
+
+import scala.concurrent.duration.FiniteDuration
+
+/**
+  * The configuration for a [[StopStrategy]]
+  *
+  * @param lapsedSinceLastInteraction   Some(duration) if the actor should stop after no new messages are received in the ''duration'' interval;
+  *                                     None to keep the actor alive
+  * @param lapsedSinceRecoveryCompleted Some(duration) if the actor should stop (and passivate);
+  *                                     None to keep the actor alive (and no passivation)
+  */
+final case class StopStrategyConfig(
```

Comment:
ADT ?

Reply:
No ADTs on configuration. It is just gonna be more difficult to generate a pureconfig reader.

Reply:
Yes but it could be nice to see directly in the conf that the never is expressed in some way in the conf other than a reference

Reply:
StopStrategyConfig has no restrictions. The configuration does not decide if it is a transient or a persistent stopStrategy, that is decide by us in the code. So it is always correct to have both parameters as optional.

Notice that in the case of `SnapshotStrategyConfig` I dealt with corner cases by having the case class private and the companion object constructor deal with the exceptions. This approach provides the safety that no wrong arguments can be passed, while keeping it simple so that pureconfig reader can be built easily